### DISCUSSION TITLE
Size-aware deduplication

### DISCUSSION THREAD

**Discussion by letienne** [ente-io/ente#3886] [Enhancements]
Hi all,

I recently migrated all my pictures from google to ente. Most of my pictures were initially "downsized" when backed up to google (and therefore their 2048x2048 version was migrated). But for some of them I still have the original stored safely on the side.

If I add the original to any album, I now have 2 "identical" pictures (same name, same exif, etc), but with different filesizes & resolution.

To my knowledge there is no way to detect/replace duplicates in this case. Ideally I would like Ente to detect that two pictures are the same, but one has a better quality than the other, and suggest to remove the "low res" version.

Is there any plan to implement such as feature? Would it be possible to use the new "Machine learning" to achieve and detect that?

Thanks ! And keep up the good work :)

**Comment by qaeu** [comment]
Related discussion with a different use case, has a staff response: #1585

**Comment by letienne** [comment]
Thanks :) However my use case is not exactly the same, as in my case the images are "exactly" the same (except compression and/or resolution)

**Comment by KestrelWild** [comment]
I'm finding myself very much in need of this too. I have original photos and Google Photos reduced versions which means I now have a ton of duplicates and seemingly no way to dedupe. At the very least, an option to dedupe by image name (without hash) would be great - with a preference to keep the larger file size version when dedupe is run.
